src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -106,13 +110,92 @@ static TimeSeriesStorage()
         }
 
 
+
         public TimeSeriesStorage(DocumentDatabase documentDatabase, Transaction tx)
         {
             _documentDatabase = documentDatabase;
             _documentsStorage = documentDatabase.DocumentsStorage;
 
             tx.CreateTree(TimeSeriesKeysSlice);
             tx.CreateTree(DeletedRangesKey);
+            tx.CreateTree(TimeSeriesStats);
+        }
+
+        private void UpdateTotalCountStats(DocumentsOperationContext context, string docId, string name, long count)
```

Comment:
I think that I'm missing calls to this from where we _delete_ values, no?

Reply:
Upon every segment update, I reduce the old count and add the new count. This valid also for deletes. Even when we remove entire segment, we replace it with an empty one.. so the same logic apply as well.